vcl/rules/user-agent.vcl

```diff
@@ -67,72 +67,39 @@ sub vcl_recv {
     # Checks for bad User-Agent
     if (
      req.http.user-agent ~ "^$"
-     || req.http.user-agent ~ "^Java"
-     || req.http.user-agent ~ "^Jakarta"
-     || req.http.user-agent ~ "IDBot"
-     || req.http.user-agent ~ "id-search"
-     || req.http.user-agent ~ "User-Agent"
-     || req.http.user-agent ~ "compatible ;"
-     || req.http.user-agent ~ "ConveraCrawler"
-     || req.http.user-agent ~ "^Mozilla$"
-     || req.http.user-agent ~ "libwww"
-     || req.http.user-agent ~ "lwp-trivial"
-     || req.http.user-agent ~ "curl"
-     || req.http.user-agent ~ "PHP/"
-     || req.http.user-agent ~ "urllib"
-     || req.http.user-agent ~ "GT:WWW"
-     || req.http.user-agent ~ "Snoopy"
-     || req.http.user-agent ~ "MFC_Tear_Sample"
-     || req.http.user-agent ~ "HTTP::Lite"
-     || req.http.user-agent ~ "PHPCrawl"
-     || req.http.user-agent ~ "URI::Fetch"
-     || req.http.user-agent ~ "Zend_Http_Client"
-     || req.http.user-agent ~ "http client"
-     || req.http.user-agent ~ "PECL::HTTP"
-     || req.http.user-agent ~ "panscient.com"
-     || req.http.user-agent ~ "IBM EVV"
-     || req.http.user-agent ~ "Bork-edition"
-     || req.http.user-agent ~ "Fetch API Request"
-     || req.http.user-agent ~ "PleaseCrawl"
+     || req.http.user-agent ~ "(?i)^java"
+     || req.http.user-agent ~ "(?i)^python"
+     || req.http.user-agent ~ "(?i)IDBot"
+     || req.http.user-agent ~ "(?i)id-search"
+     || req.http.user-agent ~ "(?i)^user-agent"
+     || req.http.user-agent ~ "(?i)compatible ;"
+     || req.http.user-agent ~ "(?i)^Mozilla$"
+     || req.http.user-agent ~ "(?i)libwww"
+     || req.http.user-agent ~ "(?i)lwp-trivial"
+     || req.http.user-agent ~ "(?i)^curl"
+     || req.http.user-agent ~ "(?i)PHP/"
+     || req.http.user-agent ~ "(?i)urllib"
+     || req.http.user-agent ~ "(?i)GT:WWW"
+     || req.http.user-agent ~ "(?i)Snoopy"
+     || req.http.user-agent ~ "(?i)MFC_Tear_Sample"
+     || req.http.user-agent ~ "(?i)HTTP::Lite"
+     || req.http.user-agent ~ "(?i)PHPCrawl"
+     || req.http.user-agent ~ "(?i)URI::Fetch"
+     || req.http.user-agent ~ "(?i)Zend_Http_Client"
+     || req.http.user-agent ~ "(?i)http client"
+     || req.http.user-agent ~ "(?i)PECL::HTTP"
+     || req.http.user-agent ~ "(?i)htmlyse.com"
+     || req.http.user-agent ~ "(?i)ecairn.com"
+     || req.http.user-agent ~ "(?i)IBM EVV"
+     || req.http.user-agent ~ "(?i)Bork-edition"
+     || req.http.user-agent ~ "(?i)Fetch API Request"
+     || req.http.user-agent ~ "(?i)PleaseCrawl"
```

Comment:
Why are these case insensitive? Have you seen varying case in the wild¿

Reply:
There is no special reason. It's just to make sure these user-agents are blocked, even if some characters are different from the case of the rule.